src/node_api.cc

```diff
@@ -2766,7 +2766,26 @@ class Work {
     Work* work = static_cast<Work*>(req->data);
 
     if (work->_complete != nullptr) {
-      work->_complete(work->_env, ConvertUVErrorCode(status), work->_data);
+      napi_env env = work->_env;
+
+      // Establish a handle scope here so that every callback doesn't have to.
+      // Also it is needed for the exception-handling below.
+      v8::HandleScope scope(env->isolate);
+
+      work->_complete(env, ConvertUVErrorCode(status), work->_data);
```

Comment:
So the main benefit then would be so that user C++ callback code doesn't have to use `napi_make_callback()` to call JS code; it could use the regular `napi_call_function()` instead?

We'd still need to keep the `napi_make_callback()` API to support advanced async scenarios though.